File: src/easy/#1137. N-th Tribonacci Number.js
// https://leetcode.com/problems/n-th-tribonacci-number/description/

function tribonacci(n) {
  let db = [0, 1, 1];

  for (let i = 3; i <= n; i++) {
    db.push(db[i - 3] + db[i - 2] + db[i - 1]);
  }

  return db[db.length - 1];
}

function tribonacci(n) {
  const dp = [0, 1, 1];

  if (n < 3) {
    return dp[n];
  }

  let i = 0;
  let sum = 0;

  while (i + 2 < n) {
    sum = dp[i] + dp[i + 1] + dp[i + 2];
    dp.push(sum);
    i++;
  }

  return sum;
}

File: src/easy/#1137. N-th Tribonacci Number.py
def tribonacci(n):
    arr = [0, 1, 1]

    if n < 3:
        return arr[n]

    i = 3

    while i <= n:
        s = sum(arr)
        arr.pop(0)
        arr.append(s)
        i = i + 1

    return arr[2]


def tribonacci(n):
    arr = [0, 1, 1]

    if n < 3:
        return arr[n]

    i = 3

    while i <= n:
        arr.append(arr[i-3] + arr[i-2] + arr[i-1])
        i = i + 1

    return arr
